### ISSUE TITLE
Should return type for `PermalinkGenerator` be `void` instead of `string`?

### ISSUE THREAD

**Issue by chris-dura** [valeriangalliat/markdown-it-anchor#115]
Hello!

## TL;DR --

The types say that my custom `PermalinkGenerator` should return a `string`:
https://github.com/valeriangalliat/markdown-it-anchor/blob/master/types/index.d.ts#L39

But, the docs for custom permalinks don't show a return type:
https://github.com/valeriangalliat/markdown-it-anchor#custom-permalink

And, it doesn't look like the built-in Permalink functions (`headerLink`, `linkInsideHeader`, `linkAfterHeader`) return strings either?

So, what `string` am I supposed to return in my custom `function renderPermalink (slug, opts, state, idx) {}`?

---

### More Details...

I'm using [VuePress](https://v2.vuepress.vuejs.org/reference/config.html#markdown-anchor), and I want to configure my own custom permalink, using the `permalink` option for `markdown-it-anchor`.

The types for `markdown-it-anchor` show that a `PermalinkGenerator` should [return a string](https://github.com/valeriangalliat/markdown-it-anchor/blob/master/types/index.d.ts#L39)

However, the docs for [custom permalinks](https://github.com/valeriangalliat/markdown-it-anchor#custom-permalink) don't show a return type, so I'm getting an ESLint error because I don't `return` a `string` in my custom function...

```ts
// .vuepress/config.ts
...
markdown: {
  anchor: {
    permalink: (slug, opts, state, idx) => {
      state.tokens.splice(idx, 0, Object.assign(new state.Token('html_block', '', 0), {
        content: `<div>test</div>`
      }));
    }
  }
},
...
```

```
(property) anchor.AnchorOptions.permalink?: anchor.PermalinkGenerator
Type '(slug: string, opts: PermalinkOptions, state: StateCore, idx: number) => void' is not assignable to type 'PermalinkGenerator'.
  Type 'void' is not assignable to type 'string'.ts(2322)
```

If I return a random `string`, that satisfies ESLint and the permalink still renders `<div>test</div>`, so should I just return an empty string?

```ts
// .vuepress/config.ts
...
markdown: {
  anchor: {
    permalink: (slug, opts, state, idx) => {
      state.tokens.splice(idx, 0, Object.assign(new state.Token('html_block', '', 0), {
        content: `<div>test</div>`
      }));

      return 'what should go here?';
    }
  }
},
...
```

**Comment by valeriangalliat** [comment]
Thanks for the detailed issue!

You're right this is a mistake with the return type for `PermalinkGenerator` and it should indeed be `void`.
It's fixed with v8.6.3. Cheers :)
